state-sync/state-sync-v2/state-sync-driver/src/metrics.rs

```diff
@@ -0,0 +1,114 @@
+    ExecutedTransactions,      // Executed a chunk of transactions.
+    SyncedAccounts,            // Wrote a chunk of accounts to storage.
+    SyncedTransactions,        // Wrote a chunk of transactions and outputs to storage.
+}
+
+impl StorageSynchronizerOperations {
+    pub fn get_label(&self) -> &'static str {
+        match self {
+            StorageSynchronizerOperations::AppliedTransactionOutputs => {
+                "applied_transaction_outputs"
+            }
+            StorageSynchronizerOperations::ExecutedTransactions => "executed_transactions",
+            StorageSynchronizerOperations::SyncedAccounts => "synced_accounts",
+            StorageSynchronizerOperations::SyncedTransactions => "synced_transactions",
+        }
+    }
+}
+
+/// Counter for state sync bootstrapper errors
+pub static BOOTSTRAPPER_ERRORS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "aptos_state_sync_bootstrapper_errors",
+        "Counters related to state sync bootstrapper errors",
+        &["error_label"]
+    )
+    .unwrap()
+});
+
+/// Counter for state sync continuous syncer errors
+pub static CONTINUOUS_SYNCER_ERRORS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "aptos_state_sync_continuous_syncer_errors",
+        "Counters related to state sync continuous syncer errors",
+        &["error_label"]
+    )
+    .unwrap()
+});
+
+/// Counters related to the state sync driver
+pub static DRIVER_COUNTERS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "aptos_state_sync_driver_counters",
+        "Counters related to the state sync driver",
+        &["label"]
+    )
+    .unwrap()
+});
+
+/// Counter for storage synchronizer errors
+pub static STORAGE_SYNCHRONIZER_ERRORS: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "aptos_state_sync_storage_synchronizer_errors",
+        "Counters related to storage synchronizer errors",
+        &["error_label"]
+    )
+    .unwrap()
+});
+
+/// Gauges related to the storage synchronizer
+pub static STORAGE_SYNCHRONIZER_GAUGES: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!(
+        "aptos_state_sync_storage_synchronizer_gauges",
+        "Gauges related to the storage synchronizer",
+        &["label"]
+    )
+    .unwrap()
+});
+
+/// Gauges for the storage synchronizer operations.
+/// Note: we keep this named "aptos_state_sync_version" to maintain backward
+/// compatibility with the metrics used by state sync v1.
+pub static STORAGE_SYNCHRONIZER_OPERATIONS: Lazy<IntGaugeVec> = Lazy::new(|| {
+    register_int_gauge_vec!(
+        "aptos_state_sync_version",
+        "The versions processed by the storage synchronizer operations",
+        &["storage_synchronizer_operation"]
+    )
+    .unwrap()
+});
+
+/// Increments the given counter with the provided label values.
+pub fn increment_counter(counter: &Lazy<IntCounterVec>, label: String) {
```

Comment:
Let's take in an `&str` since you're getting a ref in the end anyways.  Similarly with the others.